_posts/2018-10-22-release-1-5.md

```diff
@@ -0,0 +1,199 @@
+replicas.
+
+## pmemreorder
+
+Almost since the beginning of this project, we planned on implementing a tool
+which could exhaustively check consistency for all possible combination of
+persistent memory stores based on runtime binary instrumentation.
+This would give us, and our users, confirmation that the algorithms inside of
+our applications are in fact correct and fail-safe atomic.
+
+We already had two takes on this problem, with the second one being almost good
+enough for generic use. And so, after almost two years, we have finally decided
+that it's time to revisit the topic and cleanup the old codebase so that it can
+be released alongside pmemcheck.
+
+Stay tuned for a blog post about pmemreorder. In the meantime, see its
+[man page](http://pmem.io/pmdk/manpages/linux/master/pmreorder/pmreorder.1.html).
+
+## Flexible memcpy
+
+Right from the start, libpmem shipped with custom-made memcpy/memset
+implementations that made use of non-temporal stores (stores that do not go
+through the CPU cache) in a deterministic fashion, which we leveraged to
+optimize PMDK algorithms. But we've discovered optimization opportunities
+that required even more control over the behavior of these functions, and so
+we decided to add new variants of libpmem primitives (flush, memcpy, ...) that
+take flags which control how the functions behave.
+
+We found this so useful that we also exposed this functionality in libpmemobj
+variants of persistence primitives.
+
+For more information, see *pmemobj_memcpy_persist* [man page](http://pmem.io/pmdk/manpages/linux/master/libpmemobj/pmemobj_memcpy_persist.3).
+
+## libpmemobj changes
+
+Since libpmemobj is our most important library, that's where we spent most of
+our time. While there are only a few new features, the scope of changes
+is large due to our focus on transactional performance optimizations.
+
+### Lazily-initialized volatile variables
+
+The pmem synchronization primitives provided by libpmemobj are implemented using
+an internal mechanism that calls an initialization function on the first access
+to a variable. That mechanism could be useful for users wanting to employ custom
+synchronization primitives or even implement entirely custom algorithms that
+require volatile variables on persistent memory.
+
+This a fairly small API addition which has large amount of potential use cases.
+See its [man page](http://pmem.io/pmdk/manpages/linux/master/libpmemobj/oid_is_null.3) for more details.
+
+### Custom allocation class improvements
+
+In the previous release of libpmemobj, we've included a feature to create
+custom allocation class at runtime which can be used to optimize fragmentation
+and performance of the heap.
+
+The initial implementation had some limitations. The biggest of them was the
+128 byte lower boundry for allocation class size. This was problematic for
+workloads with smaller objects, we encountered that during our Java and Python
```

Comment:
*which* we encountered?

Reply:
hmm, no.
"which we noticed during?" or "which we observed during?"